src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/CommonCryptographyManager.cs

```diff
@@ -62,9 +62,14 @@ public byte[] CreateSha256HashBytes(string input)
         /// <remarks>AAD only supports RSA certs for client credentials </remarks>
         public virtual byte[] SignWithCertificate(string message, X509Certificate2 certificate)
         {
+            //Ensure certificate is of type RSA.
+            if (!certificate.PublicKey.Oid.FriendlyName.Equals(Constants.RsaFriendlyName))
```

Comment:
not sure if `Oid.FriendlyName` is the best way to do this. looking into if there is abetter way to check for RSA